src/main/scala/playpen/ws/TracingWSClientConfig.scala

```
@@ -0,0 +1,42 @@
+package playpen.ws
+
+import javax.inject.{ Inject, Provider, Singleton }
+
+import com.ning.http.client.AsyncHttpClientConfig.Builder
+import play.api.libs.ws.ning.{ NingAsyncHttpClientConfigBuilder, NingWSClientConfig }
+import play.api.{ Configuration, Environment }
+
+
+/**
+  * Tracing client config to enforce a user-agent is defined in the config
+  *
+  * @param userAgent          The user-agent to use when making a request.
+  * @param ningWSClientConfig The Ning client config.
+  */
+case class TracingWSClientConfig(userAgent: String,
+                                 ningWSClientConfig: NingWSClientConfig = NingWSClientConfig())
+
+@Singleton
+class TracingWSConfigParser @Inject()(ningWSClientConfig: NingWSClientConfig,
+                                      configuration: Configuration,
+                                      environment: Environment) extends Provider[TracingWSClientConfig] {
+
+  def get = parse()
+
+  def parse(): TracingWSClientConfig = {
+    val userAgent = configuration.underlying.getString("play.ws.useragent")
```

That should throw an exception, the idea is to enforce the user to define a user-agent, rather then allowing it to be optional like in play config
https://github.com/playframework/playframework/blob/2.4.x/framework/src/play-ws/src/main/scala/play/api/libs/ws/Config.scala#L46